ostelco-ios-client/ModelControllers/PushNotificationController.swift

```diff
@@ -35,9 +35,11 @@ class PushNotificationController: NSObject {
         super.init()
         
         UNUserNotificationCenter.current().delegate = self
-        Messaging.messaging().delegate = self
         
         if !UIApplication.isTesting {
+            // Actually get messages sent through firebase's Messaging
+            Messaging.messaging().delegate = self
```

Comment:
I think this is what's causing the firebase messaging token to get set as a real token on the mock. Not setting the delegate while testing should, in theory, mean that even if a delegate method is called, it won't go through to the mock notification controller.